linkerd/timeout/src/failfast.rs

```diff
@@ -51,6 +52,10 @@ impl<S> FailFast<S> {
             scope,
             inner,
             max_unavailable,
+            // the delay is reset whenever the service becomes unavailable; this
+            // initial one will never actually be used, so it's okay to start it
+            // now.
+            wait: Box::pin(time::sleep(max_unavailable)),
```

Comment:
wdyt about using a max-duration to make this clearer?

Reply:
[`Duration::MAX` is unstable](https://doc.rust-lang.org/std/time/struct.Duration.html#associatedconstant.MAX), so I used `Duration::default()` (which is zero) instead.